Automattic/wp-calypso
# Launchpad: Add use launchpad checklist hook to launchpad sidebar

Related to https://github.com/Automattic/wp-calypso/issues/75504

**Review time: short**
**Testing time: med**

## Proposed Changes

* Instead of using client-side definitions, fetch checklist and task definitions from the launchpad checklist API for the sidebar
* Use the fetched task definitions to render the launchpad checklist

## Screenshots

![2023-04-27 00 23 16](https://user-images.githubusercontent.com/5414230/234789646-949b7381-d8d2-4ea1-850d-20e4688bf6e5.gif)

## Testing Instructions


1. Check out this branch and run `yarn` and `yarn start` if needed. 
2. Start any new launchpad enabled site
   - Newsletter http://calypso.localhost:3000/setup/newsletter/intro 
   - Link-in-Bio http://calypso.localhost:3000/setup/link-in-bio/intro
   - VideoPress http://calypso.localhost:3000/setup/videopress/intro
   - Build http://calypso.localhost:3000/start ( select `Other` as the site goal )
   - Write http://calypso.localhost:3000/start ( select `Write & Publish` as the site goal )
3. Navigate to the Launchpad by completing onboarding
4. Verify that the number of tasks for each flow has remained the same
5. Verify that the text copy of each task for each flow has remained the same
6. Verify that the tasks for each flow behave exactly like before when clicked ( launch site button launches the site, edit site design redirects user to site editor, etc. )

## Pre-merge Checklist
